### PR TITLE
EX-1874 stratagem: iml-agent needs to be more verbose

### PR DESCRIPTION
**Pull request by beevans**
Report success and failures from iml-agent filesync and
cloudsync jobs

Signed-off-by: Ben Evans <beevans@whamcloud.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2310)

### CODE REVIEW

File: iml-agent/src/cli.rs [hunk 1]
```diff
@@ -560,7 +560,7 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
                 let fids: Vec<_> = fids.into_iter().map(Result::unwrap).collect();
                 let errors: Vec<_> = errors.into_iter().map(Result::unwrap_err).collect();
                 if !errors.is_empty() {
-                    tracing::error!("files not found, ignoring: {:?}", errors);
+                    println!("files not found, ignoring: {:?}", errors);
```
Comment (jgrund) [inline iml-agent/src/cli.rs:563]: nit:

```suggestion
                    eprintln!("files not found, ignoring: {:?}", errors);
```

File: iml-agent/src/cli.rs [hunk 2]
```diff
@@ -607,12 +629,28 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
...
+                let result =
+                    action_cloudsync::process_fids((llapi.mntpt(), task_args, fidlist)).await;
+                match result {
+                    Ok(reslist) => {
+                        if reslist.is_empty() {
+                            println!("Ok!");
+                            exit(0);
+                        }
+
+                        for err in reslist.iter() {
```
Comment (jgrund) [inline iml-agent/src/cli.rs:641]: How large might this list be?

File: iml-agent/src/cli.rs [hunk 3]
```diff
@@ -574,8 +574,30 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
...
                 };
 
-                let _result =
+                let result =
                     action_filesync::process_fids((llapi.mntpt(), task_args, fidlist)).await;
+
+                match result {
+                    Ok(reslist) => {
+                        if reslist.is_empty() {
+                            println!("Ok!");
```
Comment (jgrund) [inline iml-agent/src/cli.rs:583]:
```suggestion
                            println!("FIDs processed successfully");
```

File: iml-agent/src/cli.rs [hunk 4]
```diff
@@ -607,12 +629,28 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
...
+                    action_cloudsync::process_fids((llapi.mntpt(), task_args, fidlist)).await;
+                match result {
+                    Ok(reslist) => {
+                        if reslist.is_empty() {
+                            println!("Ok!");
+                            exit(0);
+                        }
+
+                        for err in reslist.iter() {
+                            println!(
```
Comment (jgrund) [inline iml-agent/src/cli.rs:642]:
```suggestion
                            eprintln!(
```

File: iml-agent/src/cli.rs [hunk 5]
```diff
@@ -607,12 +629,28 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
...
+                                "Failed to sync {} ({}) {}",
+                                llapi.fid2path(&err.fid)?,
+                                err.fid,
+                                err.errno
+                            );
+                        }
+                    }
+                    Err(e) => {
+                        println!("cloudsync failed {}", e);
```
Comment (jgrund) [inline iml-agent/src/cli.rs:651]:
```suggestion
                        eprintln!("cloudsync failed {}", e);
```

File: iml-agent/src/cli.rs (hunk 4 above)
Comment (beevans) [reply iml-agent/src/cli.rs:642]: Done

File: iml-agent/src/cli.rs (hunk 3 above)
Comment (beevans) [reply iml-agent/src/cli.rs:583]: Done

File: iml-agent/src/cli.rs (hunk 2 above)
Comment (beevans) [reply iml-agent/src/cli.rs:641]: As many items as the user typed in (not more)

File: iml-agent/src/cli.rs (hunk 1 above)
Comment (beevans) [reply iml-agent/src/cli.rs:563]: Done

File: iml-agent/src/cli.rs (hunk 5 above)
Comment (beevans) [reply iml-agent/src/cli.rs:651]: done

**Review by jgrund — APPROVED** [review]

**Pull request merged**